### PR TITLE
added delete button and save progress when update

### PR DESCRIPTION
**Pull request by JinweiZhu**
Now delete button is on the opportunity board. Also, when you click update, the existing data is shown automatically. I didn't write tests because these are UI change. But if we should let me know. Also, failing CI due to black doesn't seem too right for me when we can correct it immediately.

### CODE REVIEW

**Comment by JinweiZhu** [comment]
You have 2 buttons, 1 for update 1 for delete.

**Comment by Hgersten-hash** [comment]
> You have 2 buttons, 1 for update 1 for delete.

I don't think we are putting these on the opportunity board. They belong on the editing form of posting an opportunity. Does that make sense?

**Comment by JinweiZhu** [comment]
I don't think it makes a difference. We can keep it in for right now?

**Comment by Hgersten-hash** [comment]
> I don't think it makes a difference. We can keep it in for right now?

I would rather remove them since Eden is working on the filtering part of the opportunity board and I wouldn't want this to affect that.

**Comment by JinweiZhu** [comment]
I don't think the delete button will have any impact on the filtering. Besides, I don't find having delete right next to the pencil for edit a good UI layout. It's prone to misclick the icon.

**Review by dbleek — COMMENTED** [review]
Looks great! Mostly had to sift through my own changes because of the rebase :) but deletion and loading previous opportunities looks good.

File: voluncheer/nuke-and-rebuild-db.sh [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+find . -path "*/migrations/*pycache*"  -delete
+rm db.sqlite3
+#pip uninstall -y Django
+#pip install Django==3.2.8
+git checkout origin/develop -- db.sqlite3
+mv db.sqlite3 dbTemp.sqlite3
+./manage.py makemigrations
+./manage.py migrate
+#Dump data
+sqlite3 dbTemp.sqlite3 ".dump opportunityboard_category" |sqlite3 db.sqlite3
```
Comment (dbleek) [inline voluncheer/nuke-and-rebuild-db.sh:14]: Could you explain how this works? Just for my own edification.

**Comment by EdenWuyifan** [comment]
Great work done!!! Thanks for the optimizations, Wayne!!

**Pull request merged**